packages/fxa-payments-server/src/routes/Subscriptions/index.tsx

```diff
@@ -158,6 +158,17 @@ export const Subscriptions = ({
     );
   }
 
+  // If the customer has no subscriptions, redirect to the settings page
+  if (
+    customerSubscriptions.length === 0 ||
```

Comment:
Ah so, the default state of `customerSubscriptions` is an empty array (https://github.com/mozilla/fxa/blob/master/packages/fxa-payments-server/src/store/selectors.ts#L33).  That is, when Subscriptions is initially mounted, this is _always_ true.  That's why I get redirected even when I have a subscription.

I think we could change the default state to `null`.  We do need to update any code that expects `customerSubscriptions` to be an array.

Reply:
Good find! Makes total sense.

I can update the type for `customerSubscriptions` to be `Array<CustomerSubscription> | null`, but I do worry about knock on effects for that sort of change, especially since it seems like `--strictNullChecks` isn't turned on for this project.

I'm curious about [this block](https://github.com/mozilla/fxa/pull/2824/files#diff-f2c1cf4bcabdeb9f1594a4f419931657R108-R113). I'd have figured that on the initial mount the `loading` state for these would be `true` and we wouldn't reach the redirect block. Does `loading` not get set to true until after the component has rendered the first time? If that's not the case it seems a bit strange.

Would it be valuable to look down that path?

Reply:
The `loading` states won't be true until `fetchSubscriptionsRouteResources` is called (https://github.com/mozilla/fxa/pull/2824/files#diff-f2c1cf4bcabdeb9f1594a4f419931657R100).  So there's a race condition between that and the `if` here.

Allowing `null` make sense to me.  Before data is fetched, the state is really unknown, not "no subscriptions".  `customerSubscriptions` is only used a few place, it should not be a lot of work to update those lines with a null check.

Reply:
Okay cool, looks like that was it! After switching to null, I don't see the improper redirects anymore when I have a subscription to 123done.